apps/examples/testcase/ta_tc/wifi_manager/itc/itc_wifi_manager_main.c

```diff
@@ -288,6 +288,240 @@ static void itc_wifi_manager_connect_disconnect_ap_p(void)
 	TC_SUCCESS_RESULT();
 }
 
+/**
+ * @testcase         itc_wifi_manager_connect_ap_config_p
+ * @brief            check initialize, connect, de-initalization operations of wifi manager
+ * @scenario         check initialize, connect, de-initalization operations of wifi manager
+ * @apicovered       wifi_manager_init, wifi_manager_connect_ap_config, wifi_manager_deinit
+ * @precondition     none
+ * @postcondition    none
+ */
+static void itc_wifi_manager_connect_ap_config_p(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+	ret = wifi_manager_init(&wifi_callbacks);
+	TC_ASSERT_EQ("wifi_manager_init", ret, WIFI_MANAGER_SUCCESS);
+
+	wifi_manager_ap_config_s config;
+	wifi_manager_reconnect_config_s reconfig;
+	reconfig.type = WIFI_RECONN_INTERVAL;
+	reconfig.interval = 10;
+	config.ssid_length = strlen(TEST_SSID);
+	config.passphrase_length = strlen(TEST_PASSWORD);
+	strncpy(config.ssid, TEST_SSID, config.ssid_length + 1);
+	strncpy(config.passphrase, TEST_PASSWORD, config.passphrase_length + 1);
+	config.ap_auth_type = (wifi_manager_ap_auth_type_e)TEST_AUTH_TYPE;
+	config.ap_crypto_type = (wifi_manager_ap_crypto_type_e)TEST_CRYPTO_TYPE;
+	printf("AP config: %s(%d), %s(%d), %d %d\n", config.ssid, config.ssid_length, config.passphrase, \
+			config.passphrase_length, config.ap_auth_type, config.ap_crypto_type);
+
+	ret =  wifi_manager_connect_ap_config(&config, &reconfig);
+	TC_ASSERT_EQ_CLEANUP("wifi_manager_connect_ap", ret, WIFI_MANAGER_SUCCESS, wifi_manager_deinit());
+	WIFITEST_WAIT;
+
+	ret = wifi_manager_deinit();
+	TC_ASSERT_EQ("wifi_manager_deinit", ret, WIFI_MANAGER_SUCCESS);
```

Comment:
if wifi manager succeeded to connect AP, then it have to wait the disconnect event when it called wifi_manager_deinit().
please check it.

Reply:
code changes and commit